packages/compiler/core/parser.ts

```diff
@@ -710,12 +722,12 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
     if (parseOptional(Token.Equals)) {
       def = parseExpression();
     }
-    return {
+    return createNode({
```

Comment:
Could createNode be eliminated by having finishNode add `{ symbol: undefined! }`?

Reply:
Or would that waste some storage for nodes that never get symbols? Maybe have finishNode typing claim that it adds symbol in typing, which doesn't seem morally worse than createNode's cast. I'm just wondering if we can reduce the boilerplate for finalizing a node to one call.

Reply:
I felt it was a little more efficient to avoid adding the symbol property at all if we don't need it. But you're 100% right that finishNode is kind of pointless when we have createNode. createNode can do that work. Will make that change.

Reply:
I would suggest keeping the name finishNode for the thing that remains. `create` doesn't feel like something that would mutate its input and we had a significant perf gain from not copying the input when we changed to ...finishNode. I may be missing something but I think you could drop createNode and just have the typing for finishNode claim it is adding a symbol? I did feel like { ...finshNode() } came out more readable than finishNode({ ... }). But either way doesn't really matter so long as we don't copy unnecessarily.

Reply:
So what I'm wondering, in actual code, would be this:

```diff
-  function finishNode(pos: number): TextRange & { flags: NodeFlags } {
+  function finishNode(pos: number): TextRange & { flags: NodeFlags; symbol: Sym } {
     const flags = parseErrorInNextFinishedNode ? NodeFlags.ThisNodeHasError : NodeFlags.None;
     parseErrorInNextFinishedNode = false;
-    return { pos, end: previousTokenEnd, flags };
+    // cast because we don't actually assign the symbols until binding
+    return { pos, end: previousTokenEnd, flags } as any;
   }
```

And then remove createNode.